src/pages/Upload.js

```diff
@@ -403,6 +403,8 @@ const CollectionSelectorPanel = _.flow(
     ]),
     isCreating && h(NameModal, {
       thing: 'Collection',
+      validator: /^[^ /#*?\[\]]+$/, // eslint-disable-line no-useless-escape
```

Comment:
Thinking out loudly (TOL - which implies it's not a blocker): I wonder if we should use `\s` instead of a plain space in the regex so it'd match `\t\r\n\f` as well.

Reply:
Done in [c5f6edb](https://github.com/DataBiosphere/terra-ui/pull/2887/commits/c5f6edb954c7c08c536bf27ccaa945e9afc3891f).